internal/query/target.go

```diff
@@ -0,0 +1,228 @@
+package query
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/zitadel/zitadel/internal/api/call"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/query/projection"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+var (
+	targetTable = table{
+		name:          projection.TargetTable,
+		instanceIDCol: projection.TargetInstanceIDCol,
+	}
+	TargetColumnID = Column{
+		name:  projection.TargetIDCol,
+		table: targetTable,
+	}
+	TargetColumnCreationDate = Column{
+		name:  projection.TargetCreationDateCol,
+		table: targetTable,
+	}
+	TargetColumnChangeDate = Column{
+		name:  projection.TargetChangeDateCol,
+		table: targetTable,
+	}
+	TargetColumnResourceOwner = Column{
+		name:  projection.TargetResourceOwnerCol,
+		table: targetTable,
+	}
+	TargetColumnInstanceID = Column{
+		name:  projection.TargetInstanceIDCol,
+		table: targetTable,
+	}
+	TargetColumnSequence = Column{
+		name:  projection.TargetSequenceCol,
+		table: targetTable,
+	}
+	TargetColumnName = Column{
+		name:  projection.TargetNameCol,
+		table: targetTable,
+	}
+	TargetColumnTargetType = Column{
+		name:  projection.TargetTargetType,
+		table: targetTable,
+	}
+	TargetColumnURL = Column{
+		name:  projection.TargetURLCol,
+		table: targetTable,
+	}
+	TargetColumnTimeout = Column{
+		name:  projection.TargetTimeoutCol,
+		table: targetTable,
+	}
+	TargetColumnAsync = Column{
+		name:  projection.TargetAsyncCol,
+		table: targetTable,
+	}
+	TargetColumnInterruptOnError = Column{
+		name:  projection.TargetInterruptOnErrorCol,
+		table: targetTable,
+	}
+)
+
+type Targets struct {
+	SearchResponse
+	Targets []*Target
+}
+
+func (t *Targets) SetState(s *State) {
+	t.State = s
+}
+
+type Target struct {
+	ID string
+	*domain.ObjectDetails
```

Comment:
IMO i'd either have an explicit field or remove the pointer
current solution is prone for nil pointer panics